packages/jsii/lib/compiler.ts

```diff
@@ -1,1573 +1,178 @@
     }
+    return COMPILER_OPTIONS.lib.map(name => path.join(lib, name));
 }
 
-/**
- * Reads all jsii deps of a module and creates a lookup table so we can resolve
- * external types. Also returns the set of jsii specs for each module, just in case.
- *
- * @param rootDir The root dir of the module
- * @param packageDeps The 'dependencies' section of package.json
- */
-async function readDependencies(rootDir: string, packageDeps: any, bundledDeps: undefined | string[], targets: Set<string>) {
-    const lookup = new Map<string, spec.Type>();
-    const dependencies: { [dep: string]: spec.PackageVersion } = { };
-    const bundled: { [name: string]: string } = { };
-
-    bundledDeps = bundledDeps || [ ];
-    packageDeps = packageDeps || { };
-
-    // Used to make sure we don't re-load dependencies that are present multiple times in the tree closure.
-    const visited = new Set<string>();
-    /**
-     * Loads a dependency assembly, registers the types it declares in the ``lookup`` map, and adds an entry in the
-     * ``mod.dependencies`` unless the dependency is ``transitive``.
-     *
-     * @param packageName   the name of the package found in dependencies, which will be registered.
-     * @param moduleRootDir the root directory of the module that declares the dependency being loaded.
-     * @param transitive    whether this dependeency is transitive (aka it shouldn't be added to ``mod.dependencies``)
-     */
-    async function addDependency(packageName: string, moduleRootDir: string, transitive = false) {
-        // no need to re-visit transitive dependencies (also handles cycles)
-        // if this is a direct dependency, we must add it to the assembly
-        if (visited.has(packageName) && transitive) {
-            return;
-        }
-
-        const { jsii, pkg, moduleDir } = await readJsiiForModule(moduleRootDir, packageName);
-
-        // verify that dependencies specify names for all languages defined by this package
-        // this is required in order for us to be able to resolve jsii symbols in native languages.
-        for (const target of targets) {
-            if (!(target in pkg.targets)) {
-                // tslint:disable-next-line:max-line-length
-                throw new Error(`Dependent package ${packageName} does not have a configuration specified for target '${target}' which is defined by this module`);
-            }
-        }
-
-        const moduleName = jsii.name;
-        visited.add(moduleName);
-        if (!transitive) {
-            dependencies[moduleName] = {
-                version: jsii.version,
-                targets: jsii.targets,
-                dependencies: jsii.dependencies
-            };
-        }
-
-        // add all types to lookup table.
-        if (jsii.types) {
-            for (const fqn of Object.keys(jsii.types)) {
-                lookup.set(fqn, jsii.types[fqn]);
+const SOURCE_DIRS = new Set(['bin', 'lib', 'test']);
```

Comment:
this should probably be configurable (at some point). issue?

Reply:
#203